patroni/config.py

```diff
@@ -34,121 +35,162 @@
 
 
 def default_validator(conf: Dict[str, Any]) -> List[str]:
+    """Ensure *conf* is not empty.
+
+    Designed to be used as default validator for :class:`Config` objects, if no specific validator is provided.
+
+    :param conf: configuration to be validated.
+
+    :returns: an empty list -- :class:`Config` expects the validator to return a list of 0 or more issues found while
+        validating the configuration.
+
+    :raises:
+        :class:`ConfigParseError`: if *conf* is empty.
+    """
     if not conf:
         raise ConfigParseError("Config is empty.")
     return []
 
 
 class GlobalConfig(object):
+    """A class that wraps global configuration and provides convenient methods to access/check values.
 
-    """A class that wrapps global configuration and provides convinient methods to access/check values.
-
-    It is instantiated by calling :func:`Config.global_config` method which picks either a
+    It is instantiated by calling :func:`get_global_config` method which picks either a
```

Comment:
Hmm, I see. What about this?

```diff
diff --git a/patroni/config.py b/patroni/config.py
index 2115504..6dffc13 100644
--- a/patroni/config.py
+++ b/patroni/config.py
@@ -55,8 +55,8 @@ def default_validator(conf: Dict[str, Any]) -> List[str]:
 class GlobalConfig(object):
     """A class that wraps global configuration and provides convenient methods to access/check values.

-    It is instantiated by calling :func:`get_global_config` method which picks either a
-    configuration from provided :class:`Cluster` object (the most up-to-date) or from the
+    It is instantiated either by calling :func:`get_global_config` or :meth:`Config.get_global_config`, which picks
+    either a configuration from provided :class:`Cluster` object (the most up-to-date) or from the
     local cache if :class:`ClusterConfig` is not initialized or doesn't have a valid config.
     """
```

I ask because I can see there is a function `get_global_config`, which can be used directly or through method `Config.get_global_config`.

Reply:
Oh... Forgot about a function :(
Your suggestion makes sense.

Reply:
Just pushed a commit with the above diff.